samza-api/src/main/java/org/apache/samza/table/TableDescriptorsProvider.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.table;
+
+import java.util.List;
+
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.config.Config;
+import org.apache.samza.operators.TableDescriptor;
+
+
+/**
+ * Provider to create a list of {@link TableDescriptor} objects to describe one or more Samza tables. This is the
+ * mechanism for providing table support for Samza low level API.
+ *
+ * Developers writing Samza jobs using Samza table(s) should describe the table(s) by implementing
+ * TableDescriptorsProvider.
+ *
+ * Typical user code using Samza tables should look like the following:
+ *
+ * <pre>
+ * {@code
+ * public class SampleTableDescriptorsProvider implements TableDescriptorsProvider {
+ *   private ReadableTable<String, Long> remoteTable;
+ *   private ReadWriteTable<String, String> localTable;
+ *
+ *   {@code @Override}
+ *   public List<TableDescriptor> getTableDescriptors() {
+ *     List<TableDescriptor> tableDescriptors = new ArrayList<>();
+ *     final TableReadFunction readRemoteTable = (TableReadFunction) key -> null;
```

Comment:
nit: do you think using a imaginary table reader function is more clear than using the lambda which returns a null? Example: tableReadFn = new MyStoreReadFunction(); 

also: the variables should end with Function or Fn since they're not referencing tables.